tests/10-api.spec.js

```diff
@@ -11,4 +12,83 @@ describe('API', () => {
     should.exist(m);
     m.memoize.should.be.a('function');
   });
+  describe('memoize', () => {
+    it('adds an item to the cache', async () => {
+      const m = new lruMemoize.default.LruMemoize();
+      let executedTestFn = false;
+      const testFn = async () => {
+        // simulate an async task
+        await delay(100);
+        executedTestFn = true;
+        return {success: true, timestamp: Date.now()};
+      };
+      const result = await m.memoize({
+        key: 'test1',
+        fn: () => testFn()
+      });
+      should.exist(result);
+      result.success.should.be.true;
+      result.timestamp.should.be.a('number');
+      executedTestFn.should.be.true;
+
+      // the second request should use the cached promise and resolved result
+      executedTestFn = false;
+      const result2 = await m.memoize({
+        key: 'test1',
+        fn: () => testFn()
```

Comment:
you can also remove the lambda here too and just pass `testFn` in as is.

You can do this for all of your `testFn` calls as far as I can tell.

Reply:
Done.